### DISCUSSION TITLE
licensing question (GPL)

### DISCUSSION THREAD

**Discussion by Dkarag** [dotnet/runtime#89305] [Q&A] [answered]
I am writing to seek your assistance in clarifying a licensing matter related to our project.

 As part of our project, we are developing applications using C#, which are specifically intended to run on the Windows operating system. To achieve this, we are utilizing .NET Runtime 6.

During a recent scan of the .NET Runtime source code, I came across references to the GNU General Public License (GPL) (inside mono subfolder    https://github.com/dotnet/runtime/blob/79823768e461a865019d7bdf70cfe0fcbeb99288/src/mono/mono/mini/THIRD-PARTY-NOTICES.TXT#L16) . This discovery has raised concerns about the potential impact on the licensing of our software.

I would be grateful if you could provide insights into whether using .NET Runtime 6 in our applications would subject them to the GPL license.
Thank you for your time and support.

**Comment by gfoidl** [comment]
The quoted license refers to test-code, and is not part of the final shipped .NET Runtime binaries you are using. Further this belongs to the Mono-subset, and for .NET Runtime this won't be used in your product.
Given that you can use the .NET Runtime under the license terms of the product (MIT).

Obligatory note: I'm not a lawyer, but still very sure that you don't have to care about GPL in this concern.

**Reply by richlander** [reply, marked as answer]
This is correct.

We keep a (very) hard line on code that is used in the product. It has to be permissive (MIT, BSD, and friends). We have a much more open policy on tests. We'll accept nearly any code as tests provided it doesn't infringe someone's copyright (obviously).

The third party notice file at repo root is correct for product code. It doesn't include any of the third party notice content for test code, per our policy.

This is covered at https://github.com/dotnet/runtime/blob/main/docs/project/copyright.md

**Reply by Dkarag** [reply]
Thank you both for your replies.
